pulp_rpm/app/models/repository.py

```diff
@@ -169,6 +173,54 @@ class RpmDistribution(PublicationDistribution):
     """
 
     TYPE = 'rpm'
+    repository_config_file_name = 'config.repo'
+    public_key_file_name = 'public.key'
+
+    def content_handler(self, path):
+        """Serve config.repo and public.key."""
+        if path == self.repository_config_file_name:
+            val = f"""[{self.name}]
+enabled=1
+baseurl={settings.CONTENT_ORIGIN}{CONTENT_PATH_PREFIX}{self.base_path}/
+gpgcheck=0
+"""
+            repository_pk = self.publication.repository.pk
+            repository = RpmRepository.objects.get(pk=repository_pk)
+            signing_service = repository.metadata_signing_service
+            if signing_service is None:
+                val += 'repo_gpgcheck=0'
+            else:
+                val += f"""repo_gpgcheck=1
+gpgkey={settings.CONTENT_ORIGIN}{CONTENT_PATH_PREFIX}{self.base_path}/{self.public_key_file_name}
+"""
+            return Response(body=val)
+
+        if path == self.public_key_file_name:
```

Comment:
I am surprised to see this code path. I expect the public key to already be in the publication and served by the generic content handler.